src/Symfony/Component/Notifier/Bridge/LineBot/LineBotTransport.php

```diff
@@ -42,20 +44,39 @@ protected function doSend(MessageInterface $message): SentMessage
             throw new UnsupportedMessageTypeException(__CLASS__, ChatMessage::class, $message);
         }
 
+        $options = $message->getOptions();
+
+        if (null === $options) {
+            $options = LineBotOptions::fromNotification(
+                $message->getNotification()
+                    ?? new Notification($message->getSubject())
+            );
+        }
```

Comment:
This part confuses me, I think this can be removed

Reply:
`getNotification()` may return null. I create a Notification with only the subject if `getNotification()` is null.

However, you might be right since I have handled it in the following code. I will clean up this code later.

Reply:
Refactored it so it is much clearer now.